docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md

```diff
@@ -612,3 +612,76 @@ const result = createDataFrame({
 /admonition %}}
 
 It allows plugin developers to display a custom UI in the context view by implementing the `getLogRowContextUi?(row: LogRowModel, runContextQuery?: () => void, origQuery?: TQuery): React.ReactNode;` method.
+
+### Toggleable filters in Log Details
+
+{{% admonition type="note" %}} This feature is currently not supported for external plugins outside of the Grafana repo. Add support for this API in a data source by implementing the `DataSourceWith<Feature>Support` interface. {{%
+/admonition %}}
+
+The [logs details]({{< relref "../../../../explore/logs-integration/#log-details-view" >}}) component offers the possibility of adding and removing filters from the query that generated a given log line by clicking the corresponding button next to each field associated with the log line. When this interface is implemented in your data source, you get access to the following functionalities:
+
+- If a positive filter for the field and field value is present in the query, the "plus" icon is displayed as active.
+- If a positive filter is active, it can be toggled off (removed) from the source query.
+- If a positive filter is active, it can be changed to a negative equivalent by clicking on the "minus" icon.
+- A negative filter for a field and field value can be added to the source query by clicking on the corresponding icon.
+
+To implement toggleable filters support in your data source plugin, you can use the `DataSourceWithToggleableQueryFiltersSupport` API.
+
+```ts
+import {
+  queryHasPositiveFilter,
+  removePositiveFilterFromQuery,
+  addPositiveFilterToQuery,
+  addNegativeFilterToQuery,
+} from '../your/own/package/functions';
+import { DataSourceWithToggleableQueryFiltersSupport, QueryFilterOptions } from '@grafana/data';
+
+export class ExampleDatasource
+  extends DataSourceApi<ExampleQuery, ExampleOptions>
+  implements DataSourceWithToggleableQueryFiltersSupport<ExampleQuery>
+{
+  /**
+   * Given a query, determine if it has a filter that matches the options.
+   */
+  queryHasFilter(query: ExampleQuery, filter: QueryFilterOptions): boolean {
+    /**
+     * Pass the query and the key => value pair to your query-analyzing function.
+     * We only care about equality/positive filters as only those fields will be
+     * present in the log lines.
+     */
+    return queryHasPositiveFilter(query, filter.key, filter.value);
+  }
+
+  /**
+   * Toggle filters on and off from query.
+   * If the filter is already present, it should be removed.
+   * If the opposite filter is present, it should be replaced.
+   */
+  toggleQueryFilter(query: ExampleQuery, filter: ToggleFilterAction): LokiQuery {
+    const expression = query.query; // The current query expression.
```

Comment:
`expression` => `queryText` to be consistent across examples. I know that we are not 100% consistent everywhere, but I think it would be nice if all examples were as similar as possible with naming and concepts. 

For example I really really like that you renamed `addNegativeFilterToQuery` in example above.